### DISCUSSION TITLE
Error after Update to HA Core Update 2024.8.0

### DISCUSSION THREAD

**Discussion by duffmiester** [sca075/mqtt_vacuum_camera#234] [Q&A] [answered]
### Checklist

- [X] I have updated the integration to the latest version available
- [X] I have checked if the problem is already reported

### The problem

Getting this error after updating HA Core 2024.8.0

Logger: homeassistant.setup
Source: setup.py:269
First occurred: 8:36:14 AM (1 occurrences)
Last logged: 8:36:14 AM

Setup failed for custom integration 'mqtt_vacuum_camera': Requirements for mqtt_vacuum_camera not found: ['pillow==10.3.0'].

### What version of an integration has described problem?

2024.06.3

### What was the last working version of an integration?

2024.06.3

### What vacuum model do you have problems with?

Xiaomi 1C

### What type of platform you use?

Intel NUC (or generic x86_64)

### What version of Home Assistant do you use?

2024.8.0

### What type of installation are you running?

Home Assistant OS

### Logs or Errors shown in the HA snapshots (please enable the Debug Mode) text will be auto formatted to code.

```shell
Logger: homeassistant.setup
Source: setup.py:269
First occurred: 8:36:14 AM (1 occurrences)
Last logged: 8:36:14 AM

Setup failed for custom integration 'mqtt_vacuum_camera': Requirements for mqtt_vacuum_camera not found: ['pillow==10.3.0'].
```

### Function, that in your opinion is creating the issue.

None

**Comment by sca075** [comment]
@duffmiester intereating behavior of pip and Home Assistant.

Do not understand why, but an easy fix is to change in the Manifest >=10.3.0 to 10.4.0 
https://github.com/sca075/mqtt_vacuum_camera/blob/d0f0641957a7200ab67fc749406dcadd1b35cf0d/custom_components/mqtt_vacuum_camera/manifest.json#L10
When I tested it in my PI it works on the beta.. and pass also the __hass_fest_ meaning why this "==" is coming out I don't really know. If you install the Camera 2024.08.0 might work.

**Comment by sca075** [comment]
Okay I did setup also on the main instance that never used Pillow 10.4.0 and it looks Home Assistant did boot the Camera correctly so @duffmiester sorry but __you might need to update the camera too__ as you are running __2024.06.3__ to 2024.08.0 this should solve your issue please see issiue #231 the Camera was updated accordingly.

**Comment by sca075** [comment]
@duffmiester converting this "issue" to "discussion" as it isn't I guess an issue ;)

**Comment by duffmiester** [comment]
Apologies, I didn't notice I was running such an old version of the camera integration. I have updated it and it is fine now. Thanks for the quick response! :-)

**Reply by sca075** [reply, marked as answer]
Absolutely no problems @duffmiester you are always welcome to raise questions and issues here 👍

**Comment by jgkawell** [comment]
@sca075 I'm actually still having this problem after updating HA to 2024.8.1 (directly from a 2024.7.x version). I've tried redownloading the camera integration, restarting HA, restarting the HA host, and even reverting the camera integration to a v8 beta version and then back to the v8 stable. I'm at a loss as after all of that I'm still getting the same pillow version error:

```
Setup failed for custom integration 'valetudo_vacuum_camera': Requirements for valetudo_vacuum_camera not found: ['pillow==10.3.0']. 
```

Any suggestions?

**Reply by sca075** [reply]
@jgkawell what version of the camera is running on your system?

**Reply by sca075** [reply]
> @jgkawell what version of the camera is running on your system?

valetudo_vacuum_camera isn't anymore available the new versions are for mqtt_vacuum camera.

Give me some minutes and I will write you a procedure to update the camera to the newest version.

**Reply by sca075** [reply]
@jgkawell 

Okay please pay attention to the following steps.

Version 2024.06.4 was the last Valetudo_Vaccum_Camera version.

This version automatically migrates the entries in the Mqtt_vaccum_camera that of course includes the fixes for this dependency issue.

Honestly I didn't test the migration on the 2024.8.1 of Home Assistant but if you are willing to do it then:

1) please check the instructions for the manual installation and choose the version 2024.06.4,

It's easy probably to remove all files from the valetudo_vacuum_camera and unzip the files of 2024.06.4 over there if you can access HA via Samba.

2) Run the camera by rebooting HA a repair will pop up. 

3) Backup the system (HA) full backup before you start the repair please.

4) change on HACS the url for this component, delete the old valetudo _vacuum_camera if still available.

5) note there's already a MQTT_Vacuum_Camera over your system probably, you can manually configure that (above should do all for you), or if you followed till now.. then just re-update the camera to 2024.08.0

**Reply by jgkawell** [reply]
Ahh thank you for explaining this! It turned out all I had wrong was I hadn't switched the integration from the "valetudo" one to the "mqtt" one. My HACS was correct but the integration was pointed at the old name. Setting up the new "mqtt" integration worked perfectly. Thanks!
